### PR TITLE
Remove the unneeded array to object conversion code

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3607/
Test FAILed.

**Comment by jaycee** [comment]
QA:

Ensure bundle stuff works -- details and deployment specifically.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js [hunk 1]
```diff
@@ -272,22 +272,9 @@ var module = module;
           newKey = key.toLowerCase();
         }
         // Create shallow copies of objects for the copy.
-        var copy;
-        // XXX jcsackett 2015-11-09: Previously we used Y.merge to make shallow
-        // copies. As a side effect, arrays were munged into weird objects.
-        // Unfortunately the GUI relies on this side effect. When we can update
```
Comment (jaycee) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: Turns out, no it does not--only the tests rely on the side effect.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3612/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA :ok: Thanks for this investigation!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1227

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**